### DISCUSSION TITLE
Repeating macro.

### DISCUSSION THREAD

**Discussion by Plyply99** [sezanzeb/input-remapper#367] [Q&A]
Greetings all,
Just learning linux for couple weeks now and found this amazing program. I'm using it to make macros for my keyboard and mouse.
Question I have, is there a way to make a macro repeat itself with a toggle key?
Example:  Press key "A"
                 Macro runs repeatedly
                 Press key "A" again
                 Macro stops.

In Autohotkey I would use the Toggle command, but not sure how to do it in Linux.

**Comment by sezanzeb** [comment]
hey, sorry for taking my time to answer.

I think as of now this is actually not possible. I moved this into a new issue: https://github.com/sezanzeb/input-remapper/issues/378

**Reply by Plyply99** [reply]
Thank you!

**Comment by nullbyto** [comment]
You could do something like this (which will toggle a single key not a macro):
```
if_eq($flag, 0, set(flag, 1), set(flag, 0)).
if_eq($flag, 1, key_down(KEY_A), key_up(KEY_A))
```

This should be added as a macro to toggle a key which will make it easier.
```
toggle(KEY_A)
```

**Reply by sezanzeb** [reply]
Thanks for the suggestion. Here is a pull request that adds `toggle`, if you want to try it before I merge it: https://github.com/sezanzeb/input-remapper/pull/427
